src/components/ChangePasswordForm.js

```diff
@@ -0,0 +1,101 @@
+// @flow
+
+import * as React from 'react';
+
+import { observer } from 'mobx-react';
+import AuthApi from '../AuthApi';
+import InjectedComponent from '../common/InjectedComponent';
+import { FormValidationMessage } from 'react-native-elements';
+import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
+import { form, FormField } from '../common/form';
+import SectionContent from '../common/SectionContent';
+import TextField from '../components/TextField';
+import Button from '../common/buttons/Button';
+
+export type ChangePasswordFormFields = {|
+  confirmPassword: string,
+  newPassword: string,
+  oldPassword: string,
+|};
+
+const validate = ({
+  confirmPassword,
+  newPassword,
+  oldPassword,
+}: ChangePasswordFormFields): { [key: string]: string } => {
+  const errors = {};
+
+  if (!oldPassword) {
+    errors.oldPassword = 'Old password is required';
+  }
+
+  if (!newPassword) {
+    errors.newPassword = 'New password is required';
+  }
+
+  if (newPassword && newPassword === oldPassword) {
+    errors.newPassword = 'New password the same as old';
+  }
+
+  if (newPassword !== confirmPassword) {
```

Comment:
Don't put a confirm password. 

All the apps on mobile moved away from the password + confirm password because it's a pain to type on mobile.  If you remove the confirm password, it makes it way easier for people to sign up.